interchaintest/feegrant_test.go

```diff
@@ -0,0 +1,530 @@
+				sdkConf.SetBech32PrefixForAccount(cProv.PCfg.AccountPrefix, cProv.PCfg.AccountPrefix+"pub")
+				sdkConf.SetBech32PrefixForValidator(cProv.PCfg.AccountPrefix+"valoper", cProv.PCfg.AccountPrefix+"valoperpub")
+				sdkConf.SetBech32PrefixForConsensusNode(cProv.PCfg.AccountPrefix+"valcons", cProv.PCfg.AccountPrefix+"valconspub")
+
+				hash, err := hex.DecodeString(curr.Response.TxHash)
+				require.Nil(t, err)
+				txResp, err := cProv.RPCClient.Tx(ctx, hash, true)
+				require.Nil(t, err)
+
+				require.Nil(t, err)
+				dc := cProv.Cdc.TxConfig.TxDecoder()
+				tx, err := dc(txResp.Tx)
+				require.Nil(t, err)
+				builder, err := cProv.Cdc.TxConfig.WrapTxBuilder(tx)
+				require.Nil(t, err)
+				txFinder := builder.(protoTxProvider)
+				fullTx := txFinder.GetProtoTx()
+				isFeegrantedMsg := false
+
+				msgType := ""
+				for _, m := range fullTx.GetMsgs() {
+					msgType = types.MsgTypeURL(m)
+					//We want all IBC transfers (on an open channel/connection) to be feegranted in round robin fashion
+					if msgType == "/ibc.core.channel.v1.MsgRecvPacket" || msgType == "/ibc.core.channel.v1.MsgAcknowledgement" {
+						isFeegrantedMsg = true
+						break
+					}
+				}
+
+				//It's required that TXs be feegranted in a round robin fashion for this chain and message type
+				if isFeegrantedChain && isFeegrantedMsg {
+					signers := fullTx.GetSigners()
+					require.Equal(t, len(signers), 1)
+					granter := fullTx.FeeGranter()
+
+					//Feegranter for the TX that was signed on chain must be the relayer chain's configured feegranter
+					require.Equal(t, granter.String(), feegrantInfo.granter)
+					require.NotEmpty(t, granter)
+
+					for _, msg := range fullTx.GetMsgs() {
+						msgType = types.MsgTypeURL(msg)
+						//We want all IBC transfers (on an open channel/connection) to be feegranted in round robin fashion
+						if msgType == "/ibc.core.channel.v1.MsgRecvPacket" {
+							c := msg.(*chantypes.MsgRecvPacket)
+							appData := c.Packet.GetData()
+							tokenTransfer := &ibcgo.FungibleTokenPacketData{}
+							err := tokenTransfer.Unmarshal(appData)
+							if err == nil {
+								fmt.Printf("%+v\n", tokenTransfer)
+							} else {
+								fmt.Println(string(appData))
+							}
+						}
+					}
+
+					//Grantee for the TX that was signed on chain must be a configured grantee in the relayer's chain feegrants.
+					//In addition, the grantee must be used in round robin fashion
+					//expectedGrantee := nextGrantee(feegrantInfo)
+					actualGrantee := signers[0].String()
+					signerList, ok := feegrantMsgSigners[chain]
+					if ok {
+						signerList = append(signerList, actualGrantee)
+						feegrantMsgSigners[chain] = signerList
+					} else {
+						feegrantMsgSigners[chain] = []string{actualGrantee}
+					}
+					fmt.Printf("Chain: %s, msg type: %s, height: %d, signer: %s, granter: %s\n", chain, msgType, curr.Response.Height, actualGrantee, granter.String())
+					//require.Equal(t, expectedGrantee, actualGrantee)
```

Comment:
seems like this should be uncommented?

Reply:
I'll delete it. The correct checks are in place below that, in the for loop `for chain, signers := range feegrantMsgSigners`